cmd/fetch.go

```diff
@@ -42,30 +42,31 @@ func fetchCmd() *cobra.Command {
 		Long:   "Fetch SBOM file(s) from HTTP(S), OCI, or Git URLs",
 		PreRun: preRun(opts.Options),
 		Run: func(_ *cobra.Command, args []string) {
-			if string(outputFile) != "" {
+			var outputFile *os.File
+
+			if string(outputFileName) != "" {
 				if len(args) > 1 {
 					opts.Logger.Fatal("The --output-file option cannot be used when more than one URL is provided.")
 				}
 
-				out, err := os.Create(string(outputFile))
+				out, err := os.Create(string(outputFileName))
 				if err != nil {
 					opts.Logger.Fatal("error creating output file", "outputFile", outputFile)
 				}
 
-				opts.OutputFile = out
-
-				defer opts.OutputFile.Close()
+				outputFile = out
+				defer outputFile.Close()
 			}
 
 			for _, url := range args {
-				if err := fetch.Fetch(url, opts); err != nil {
+				if err := fetch.Fetch(url, opts, outputFile); err != nil {
```

Comment:
I think the variable name change is a good idea and makes the code clearer.

Why are we bypassing the `opts.OutputFile` though? The goal of using the options struct is so that the parameter sets of entry point functions like `Fetch` don't change.

Since this PR doesn't remove `OutputFile` from the `FetchOptions` struct, was this perhaps unintentional?

Reply:
I'll have to give it some more thought, but I think what makes the most sense to me would be creating a new instance of `FetchOptions` with the new `OutputFile` on it to keep the function signature the same.

Basically recreate the setup process of when `Fetch` was originally called from `cmd/fetch.go`.